spec/controllers/requests_controller_spec.rb

```diff
@@ -2,29 +2,42 @@
 require "active_model/model"
 
 describe RequestsController, type: :controller do
-  class TestRequest
-    include ActiveModel::Model
-    include Support::Requests::WithRequester
+  let(:test_request_class) do
+    Class.new do
+      include ActiveModel::Model
+      include Support::Requests::WithRequester
 
-    attr_accessor :a, :b
+      attr_accessor :a, :b
 
-    validates_presence_of :a
+      validates_presence_of :a
+    end
   end
 
-  class TestZendeskTicket < Zendesk::ZendeskTicket
-    def subject
-      "Test request"
-    end
+  let(:test_zendesk_ticket_class) do
+    Class.new(Zendesk::ZendeskTicket) do
+      def subject
+        "Test request"
+      end
 
-    def tags
-      %w[tag_a tag_b]
-    end
+      def tags
+        %w[tag_a tag_b]
+      end
 
-    def comment_snippets
-      []
+      def comment_snippets
+        []
+      end
     end
   end
 
+  before do
+    stub_const("TestRequest", test_request_class)
```

Comment:
Optional: could we use the `let` variables directly instead of stubbing the constants? That would be slightly simpler, although I appreciate doing it this way arguably minimises the diff.

Reply:
That didn't work when I tried it as the lets aren't in scope for the [`controller do`](https://github.com/alphagov/support/pull/885/files#diff-3fdb0871f88db3cca057fda333800ff439e49d4b439456abd4b2c734b039ec12R41).

I could only get them to work as globally available constants.

Reply:
Ah yes I see. 

I had a quick go myself and it looks like that whole block is executed in some other context.